pkg/provisioning/service.go

```diff
@@ -102,6 +102,20 @@ func (s *Service) Init(ctx context.Context) error {
 		allPls = append(allPls, duplicateID)
 	}
 
+	// remove pipelines with duplicate IDs from API pipelines
+	var existingPipelinesFromAPI []int
+	for i, pl := range configs {
+		_, err := s.pipelineService.Get(ctx, pl.ID)
+		// if err is nil then a pipeline exists already, so we add towards duplicate ID list
```

Comment:
It's fine if the pipeline already exists, as long as it was provisioned using a config file. If it was provisioned by anything else, we don't want to mutate it.
```suggestion
		pl, err := s.pipelineService.Get(ctx, pl.ID)
		// check error ...
		if pl.ProvisionedBy == pipeline.ProvisionTypeConfig {
			continue // all good
		}
		// Not ok! We don't want to provision this pipeline, it was provisioned by something else.
		// ...
```